### PR TITLE
APDS9960 driver

### PR DESCRIPTION
**Pull request by Archonic944**
Yippee

- Enable the sensor
- Enable the proximity detection
- Set the proximity high and low threshold
- Set the proximity persistence filter
- Enable the proximity interrupts
- Clear the interrupts
- Disable the sensor

### CODE REVIEW

File: src/main/java/frc/utils/APDS9960.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package frc.utils;
+
+import edu.wpi.first.wpilibj.I2C;
+import frc.constants.VisionConstants;
+
+public class APDS9960 {
+  I2C i2c = new I2C(I2C.Port.kOnboard, VisionConstants.APDS9960_I2C_ADDRESS);
```
Comment (BlueZeeKing) [inline src/main/java/frc/utils/APDS9960.java]: It uses the MXP port, not the onboard one

**Review by BlueZeeKing — CHANGES_REQUESTED** [review]
Everything but the port looks good

File: src/main/java/frc/utils/APDS9960.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  /**
+   *
+   * @param reg The register you'd like to write to
+   * @param value What you'd like to write, should be in the correct position e.g. 0b10100000 if you are writing 1010 to the first four bits
+   * @param mask Should be 0b11110000 if you are overwriting the first four bits
+   */
+  private void overwrite (byte reg, byte value, byte mask) {
+    byte[] read = new byte[1];
+    i2c.read(reg, 1, read);
+    i2c.write(reg, (byte) ((read[0] & ~mask) | value));
```
Comment (BlueZeeKing) [inline src/main/java/frc/utils/APDS9960.java]: You should also mask the value to ensure you're only writing to the bits in the mask

**Review by BlueZeeKing — CHANGES_REQUESTED** [review]
Nice job catching the mistake with using setAndPreserve, but the port is still incorrect, and overwrite doesn't prevent other bits in the register from being modified.

**Comment by Archonic944** [comment]
Mb, I  just assumed that anyone writing wasn't going to set bits outside of their mask

**Comment by BlueZeeKing** [comment]
> Mb, I just assumed that anyone writing wasn't going to set bits outside of their mask

This *should* be a correct assumption, but people make mistakes and might accidentally write outside of the mask.

**Review by BlueZeeKing — APPROVED** [review]
LGTM

**Pull request merged**